### PR TITLE
🚀  Release v6.5.0-rc.4

### PR DESCRIPTION
**Pull request by ncoden**
# Foundation v6.5.0 Release Candidate 4

This release provide additional bug fixes and various patches to improve the stability of Foundation.

### Note about this pull request
As this version was built to have the best compatibility with `v6.4.4-rc1`, we used a dedicated branch and manually picked safe changes from the development branch `develop`, excluding API evolutions and breaking changes. These imported changes are the `Use pull request #xxx` commits you can see in this pull request. Development will continue on `develop` and will lead to a future release with all changes, including new features and breaking changes.

### Release Checklist
- [x] Bump versions
- [x] Update dist files
- [x] Update SCSS settings file
▾
- [x] Update documentation
▾
- [x] Prepare release for npm
- [x] Prepare release for Composer
- [x] Prepare release for Rails
  * ⚠️  Must be published on https://github.com/zurb/foundation-rails
- [x] ~~Prepare release for Meteor~~ Do not support pre-releases
  * ⚠️  Must be published before the Git release for `.versions`
- [x] Prepare release for NuGet
  * ⚠️  Must be published with https://www.npmjs.com/package/nuget-publish
▾
- [ ] Released!

## Changes from v6.5.0-rc.3

* 🐛  <samp>#11510</samp> - Follow relative and absolute links in SmoothScroll (@ncoden, #11506)
* 🐛  <samp>#11519</samp> - Clean SmoothScroll listeners on destroy (@ncoden)
* 🐛  <samp>#11529</samp> - Allow Interchange filenames to contain brackets (@peterquentin)
* 🐛  <samp>#11530</samp> - Fix various bugs with Tabs/Accordion deep-linking (@ncoden, #11527)
* 🐛  <samp>#11531</samp> - Fix various bugs with "Tap" events in Touch utility (@ncoden, #11423)
* 🚚  <samp>#11508</samp> - Provide CJS, ESM and ES6+ bundles (@ncoden, #11502)
* 📦  <samp>#11512</samp> - Update dependencies (2018-09-22) (@ncoden)
* 📦  <samp>#11528</samp> - Update and fix `foundation-docs` dependency (@ncoden)
* 📖  <samp>#11500</samp> - Improve description of the Prototype page (@chrisdfe)
* 📖  <samp>#11518</samp> - Fix typos in the Javascript documentation (@ncoden)
* 📖  <samp>#11526</samp> - Use correct namespace for DropdownMenu/OffCanvas events (@ncoden, #11523)

## Contributors
* Christopher Ferris ([@chrisdfe](https://github.com/chrisdfe)) [1]
* Nicolas Coden ([@ncoden](https://github.com/ncoden)) [23]
* Peter Quentin ([@peterquentin](https://github.com/peterquentin)) [1]

Special thanks to:
* Daniel Ruf ([@DanielRuf](https://github.com/DanielRuf))
* Kai Falkowski ([@SassNinja](https://github.com/SassNinja))

### CODE REVIEW

**Comment by ncoden** [comment]
This should be the last release candidate for Foundation v6.5.0, making it stable and usable in any enviromnent. Others known bugs will be fixed in later path releases.

@SassNinja @IamManchanda 👋 Could I have you review on this ? Thanks!

File: docs/pages/javascript.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Know that they all require `foundation.core.js` to be loaded *first*. Some plugi
 
 ### Import in JavaScript
 
-Foundation is exported as [UMD modules](http://bob.yexley.net/umd-javascript-that-runs-anywhere/). This means that Foundation and its plugins can be imported in any JavaScript environnement.
+By default, Foundation is exported as [UMD modules](http://bob.yexley.net/umd-javascript-that-runs-anywhere/). This means that Foundation and its plugins can be imported and used in any JavaScript environnement.
```
Comment (DanielRuf) [inline docs/pages/javascript.md]: environnement => environment (probably oversaw this one)

File: docs/pages/javascript.md [hunk 2]
```diff
@@ -63,6 +63,22 @@ var Foundation = require('foundation-sites');
 var $dropdown = new Dropdown('#mydropdown');
 ```
 
+#### Available formats
+
+Foundation is proposed in bundles of various module formats so you can pick the one that match the best your needs. If you don't know these terms yet, take a look at this [10-minute introduction to module formats in JavaScript](https://www.jvandemo.com/a-10-minute-primer-to-javascript-modules-module-formats-module-loaders-and-module-bundlers/). You will find in the `dist/js` folder the following bundles:
```
Comment (DanielRuf) [inline docs/pages/javascript.md]: proposed => provided (I guess you mean this?)

Comment (DanielRuf) [reply docs/pages/javascript.md]: match the best => matches your needs the best

File: docs/pages/javascript.md [hunk 3]
```diff
@@ -63,6 +63,22 @@ var Foundation = require('foundation-sites');
...
+
+* `foundation.js` <span class="label secondary">UMD</span> <span class="label">Default</span><br>
+  Compatible with most environments and tools (AMD, CJS, ESM...). It works almost everywhere by checking the module format of your environments and then using it, which make the bundle a little heavier.
```
Comment (DanielRuf) [inline docs/pages/javascript.md]: which make => which makes

File: docs/pages/javascript.md [hunk 4]
```diff
@@ -63,6 +63,22 @@ var Foundation = require('foundation-sites');
...
+
+* `foundation.cjs.js` <span class="label secondary">CommonJS</span><br>
+  Dedicated to Node.js and older bundlers like Browserify or Webpack v1.
+
+* `foundation.esm.js` <span class="label secondary">ES6 Modules</span>  (`module` in `package.json`)<br>
+  Everything is transpiled to ES5 but the modules. Dedicated to modern bundlers, like Webpack 2+ or Rollup. They will automatically use this bundle and parse the ES6 modules to remove the unused code (see [tree shaking](#tree-shaking) below).
+
+* `foundation.es6.js` <span class="label secondary">ES6</span> (`esnext` in `package.json`)<br>
+  Unlike the others bundles, this bundle is not transpiled. It contains all the Foundation sources in ES6 in a single file. Use it if you want to manually transpile it for your own targets.
```
Comment (DanielRuf) [inline docs/pages/javascript.md]: others bundles => other bundles

File: gulp/tasks/deploy.js [hunk 5]
```diff
@@ -93,6 +95,23 @@ gulp.task('deploy:dist:files', function() {
       .pipe(tsFilter.restore);
 });
 
+//
+// Generates JS bundles and puts them in the dist/ folder.
+//
+// In addition to the UMD bundle coming from the build task, the following
+// formats are generared: CJS, ESM, ES6.
```
Comment (DanielRuf) [inline gulp/tasks/deploy.js]: generared => generated

File: gulp/tasks/deploy.js [hunk 6]
```diff
@@ -93,6 +95,23 @@ gulp.task('deploy:dist:files', function() {
...
+// See "rollup.config.js" for more informations.
```
Comment (DanielRuf) [inline gulp/tasks/deploy.js]: informations => information (there is no plural form of information, same for data)

File: js/foundation.smoothScroll.js [hunk 7]
```diff
@@ -48,10 +48,8 @@ class SmoothScroll extends Plugin {
 
         // click handler function.
         var handleLinkClick = function(e) {
-            // exit function if the event source isn't coming from an anchor with href attribute starts with '#'
-            if(!$(this).is('a[href^="#"]'))  {
-                return false;
-            }
+            // Follow the link it does not point to an anchor.
```
Comment (DanielRuf) [inline js/foundation.smoothScroll.js]: it => if it

Probably better: if it is not an anchor

File: rollup.config.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        ["@babel/preset-env", { modules: false }]
+      ],
+    }),
+  ],
+};
+
+module.exports = [
+
+  // UMD
+  // Compatible with most environements and tools (AMD, CJS, ESM...),
```
Comment (DanielRuf) [inline rollup.config.js]: environements => environments

**Review by DanielRuf — DISMISSED** [review]

**Comment by ncoden** [comment]
Ahah. Thank you @DanielRuf for the typo review. I paid attention to avoid mistakes like this and even used a corrector when working on https://github.com/zurb/foundation-sites/pull/11508, but I guess that there is no better corrector than Daniel 😄.

File: rollup.config.js (hunk 8 above)
Comment (ncoden) [reply rollup.config.js]: Done :)

**Comment by ncoden** [comment]
@DanielRuf Fixed in #11518, and updated the pull request.

**Comment by DanielRuf** [comment]
A few places seem to be not updated. If you are unsure which files, I can list them. Let me know when I can review again.

File: js/foundation.smoothScroll.js (hunk 7 above)
Comment (ncoden) [reply js/foundation.smoothScroll.js]: > Probably better: if it is not an anchor

There is an hyperlink, and what the link points to, an anchor. We are not testing if the hyperlink is an anchor (with `href="#xxx"`) but it points to it.

MDN and W3C calls these elements "Anchor Element" and "fragment id", but everywhere else the "boorkmark" with `href="#xxx"` is called an anchor, and I never heard of "fragment id" before today 😄

**Comment by ncoden** [comment]
@DanielRuf Could you review this again please ?

**Comment by ncoden** [comment]
@DanielRuf 👋 Could you review this again please?

**Comment by ncoden** [comment]
@DanielRuf Will be merged tomorrow, with or without your final review 😉

**Pull request merged**